app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessNameController.scala

```diff
@@ -0,0 +1,78 @@
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
+import play.twirl.api.Html
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.IncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessNameForm
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessNameForm._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.play.language.LanguageUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BusinessNameController @Inject()(mcc: MessagesControllerComponents, incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
+                                       authService: AuthService, val languageUtils: LanguageUtils)
+                                      (implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc) with I18nSupport {
+
+  implicit class FormUtil[T](form: Form[T]) {
+    def fill(data: Option[T]): Form[T] = data.fold(form)(form.fill)
+  }
+
+  def view(businessNameForm: Form[BusinessNameModel])(implicit request: Request[AnyContent]): Html =
+    uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.business_name(
+      businessNameForm = businessNameForm,
+      postAction = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessNameController.submit(),
+      backUrl = backUrl()
+    )
+
+  def show(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised(){
+      incomeTaxSubscriptionConnector.getSelfEmployments[BusinessNameModel](BusinessNameController.businessNameKey).map{
+        case Right(businessNameData) =>
+          Ok(view(businessNameValidationForm.fill(businessNameData)))
+        case error  => throw new InternalServerException(error.toString)
+      }}
+  }
+
+  def submit(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised(){
+      businessNameValidationForm.bindFromRequest.fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors))),
+        businessNameData =>
+          incomeTaxSubscriptionConnector.saveSelfEmployments[BusinessNameModel](BusinessNameController.businessNameKey,businessNameData) map (_ =>
+            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessNameController.show()))
+      )
+    }}
+
+  def backUrl(): String =
+    uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show().url
+}
+
+object BusinessNameController{
+  val businessNameKey: String = "businessName"
```

Comment:
```suggestion
  val businessNameKey: String = "BusinessName"
```
Keeping it consistent with the other controller